polyfill/lib/yearmonth.mjs

```diff
@@ -77,10 +78,11 @@ export class YearMonth {
   minus(durationLike, disambiguation = 'constrain') {
     if (!ES.IsYearMonth(this)) throw new TypeError('invalid receiver');
     disambiguation = ES.ToArithmeticDisambiguation(disambiguation);
-    const duration = ES.ToLimitedDuration(durationLike, [DAYS, HOURS, MINUTES, SECONDS, MILLISECONDS, MICROSECONDS, NANOSECONDS]);
+    const duration = ES.ToLimitedDuration(durationLike);
     let { year, month } = this;
-    const { years, months } = duration;
-    ({ year, month } = ES.SubtractDate(year, month, 1, years, months, 0, disambiguation));
+    const { years, months, hours, minutes, seconds, milliseconds, microseconds, nanoseconds } = duration;
+    const { days } = ES.BalanceDuration(duration.days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds, 'days');
+    ({ year, month } = ES.SubtractDate(year, month, 1, years, months, days, disambiguation));
```

Comment:
There's one potential problem here.  Let's say you have a Duration that balances to 5 days.  When you add it to a YearMonth, you get the same YearMonth (assuming no month is < 5 days long), but when you subtract it, you get the previous YearMonth, meaning that arithmetic is not symmetric.

Not sure about the best solution.  Probably it's an edge case we can just ignore.

Reply:
Hmm. It doesn't seem all that important but then again it should be consistent with other types, and we have e.g. `Temporal.Date.from('2020-03-12').minus({hours: 5})` equals 2020-03-12, so it's not currently consistent.

Maybe for subtraction we should subtract the date from the last day of the month. I'll revise this PR to do that.